src/cljc/matcher_combinators/cljs_test.cljc

```diff
@@ -9,6 +9,18 @@
             [matcher-combinators.result :as result]
             [cljs.test :as t :refer-macros [deftest is]]))
 
+(defrecord CljsMismatch [summary match-result])
+
+(defn tagged-for-pretty-printing [actual-summary result]
+  (->CljsMismatch actual-summary result))
+
+#?(:cljs
+   (extend-protocol IPrintWithWriter
+     CljsMismatch
+     (-pr-writer [cljs-mismatch writer _]
+       (-write writer (printer/as-string (-> cljs-mismatch :match-result ::result/value))))))
```

Comment:
is there a way to do this without going through `printer/as-string`? Something similar to the clj version that does:
```clojure
(defmethod clojure.core/print-method ::mismatch [{:keys [match-result]} out]
  (binding [*out* out]
    (printer/pretty-print (::result/value match-result))))
```